ml/src/main/scala/ai/h2o/sparkling/ml/models/H2OTargetEncoderModel.scala

```diff
@@ -31,16 +32,22 @@ import water.api.schemas3.KeyV3.FrameKeyV3
 
 import scala.collection.JavaConverters._
 
-class H2OTargetEncoderModel(override val uid: String, targetEncoderModel: H2OModel)
+class H2OTargetEncoderModel(override val uid: String, targetEncoderModel: Option[H2OModel])
```

Comment:
H2O backend can throw the following exception during the model training:
```
    >                       format(target_id, ".", name), value)
     E                   py4j.protocol.Py4JJavaError: An error occurred while calling o6687.fit.
     E                   : java.lang.Exception: H2O Job $0301c0a8016732d4ffffffff$_9012a5666c1e03e6f43e8238aedd4034 has failed!
     E                   Exception: water.exceptions.H2OModelBuilderIllegalArgumentException: Illegal argument(s) for TargetEncoder model: TargetEncoder_model_Java_1618311411921_21.  Details: ERRR on field: _train: Training data must have at least 2 features (incl. response).
     E                   
     E                   StackTrace: water.exceptions.H2OModelBuilderIllegalArgumentException: Illegal argument(s) for TargetEncoder model: TargetEncoder_model_Java_1618311411921_21.  Details: ERRR on field: _train: Training data must have at least 2 features (incl. response).
     E                   
     E                      at water.exceptions.H2OModelBuilderIllegalArgumentException.makeFromBuilder(H2OModelBuilderIllegalArgumentException.java:19)
     E                      at ai.h2o.targetencoding.TargetEncoder$TargetEncoderDriver.computeImpl(TargetEncoder.java:125)
     E                      at hex.ModelBuilder$Driver.compute2(ModelBuilder.java:246)
     E                      at water.H2O$H2OCountedCompleter.compute(H2O.java:1600)
     E                      at jsr166y.CountedCompleter.exec(CountedCompleter.java:468)
     E                      at jsr166y.ForkJoinTask.doExec(ForkJoinTask.java:263)
     E                      at jsr166y.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:974)
     E                      at jsr166y.ForkJoinPool.runWorker(ForkJoinPool.java:1477)
     E                      at jsr166y.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:104)
     E                   
     E                      at ai.h2o.sparkling.backend.H2OJob.waitForFinish(H2OJob.scala:39)
     E                      at ai.h2o.sparkling.ml.utils.EstimatorCommonUtils.trainAndGetDestinationKey(EstimatorCommonUtils.scala:43)
     E                      at ai.h2o.sparkling.ml.utils.EstimatorCommonUtils.trainAndGetDestinationKey$(EstimatorCommonUtils.scala:29)
     E                      at ai.h2o.sparkling.ml.features.H2OTargetEncoder.trainAndGetDestinationKey(H2OTargetEncoder.scala:34)
     E                      at ai.h2o.sparkling.ml.features.H2OTargetEncoder.fit(H2OTargetEncoder.scala:70)
     E                      at sun.reflect.GeneratedMethodAccessor217.invoke(Unknown Source)
     E                      at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     E                      at java.lang.reflect.Method.invoke(Method.java:498)
     E                      at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
     E                      at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
     E                      at py4j.Gateway.invoke(Gateway.java:282)
     E                      at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
     E                      at py4j.commands.CallCommand.execute(CallCommand.java:79)
     E                      at py4j.GatewayConnection.run(GatewayConnection.java:238)
     E                      at java.lang.Thread.run(Thread.java:748)
     ```
     
   So if the input columns are empty, I don't try to train a model and `targetEncoderModel` is None

Reply:
yes, currently works as expected: 
- if `columns_to_encode` is None, then TE will autodetect categoricals and possibly behave as no-op if there's nothing to encode.
- if `columns_to_encode` is set, TE automatically removes the other columns (I think it's an inconsistency in TE's behaviour as it should relay only on X to remove columns from train dataset, and I will probably change this in the future).
-  if `columns_to_encode` is set and empty, as described above, all predictors will be remove from train and `ModelBuilder` validation raises the error above: I also have a fix so that this validation behaves consistenly and take only predictors into consideration (currently it counts fold/weights/offset columns as predictors...).